plugin/pkg/scheduler/algorithm/priorities/util/non_zero.go

```diff
@@ -48,3 +53,68 @@ func GetNonzeroRequests(requests *api.ResourceList) (int64, int64) {
 	}
 	return outMilliCPU, outMemory
 }
+
+// FilterPodsByNameSpaces filters the pods based the given list of namespaces
+func FilterPodsByNameSpaces(names sets.String, pods []*api.Pod) []*api.Pod {
+	if len(pods) == 0 || len(names) == 0 {
+		return pods
+	}
+	result := []*api.Pod{}
+	for _, pod := range pods {
+		if names.Has(pod.Namespace) {
+			result = append(result, pod)
+		}
+	}
+	return result
+}
+
+// GetNamespacesFromPodAffinityTerm returns a set of names
+// according to the namespaces indicated in podAffinityTerm.
+// if the NameSpaces is nil considers the given pod's namespace
+// if the Namespaces is empty list then considers all the namespaces
+func GetNamespacesFromPodAffinityTerm(pod *api.Pod, podAffinityTerm api.PodAffinityTerm) sets.String {
+	names := sets.String{}
+	if podAffinityTerm.Namespaces == nil {
+		names.Insert(pod.Namespace)
+	} else if len(podAffinityTerm.Namespaces) != 0 {
+		for _, nameSpace := range podAffinityTerm.Namespaces {
+			names.Insert(nameSpace.Name)
+		}
+	}
+	return names
+}
+
+// NodeHasTopologyKey checks if nodeA and nodeB have same label value with given topologyKey as label key.
+// If the topologyKey is nil/empty, regard as the two nodes have same topologykey and label value.
+func NodesHaveSameTopologyKey(nodeA *api.Node, nodeB *api.Node, topologyKey string) bool {
+	if len(topologyKey) == 0 {
```

Comment:
(Note that I updated the above comment after I wrote it, so please read it from the web site, not email.)

Also, one other thing I just realized. I don't think we should fix it in this PR, but we should add a TODO to fix it later. 

Say you are using soft anti-affinity with empty topologyKey (this is the only case where empty topologyKey is allowed). Say you have two pods that are different for all topology keys. We will count that once (with whatever Weight is specified in the corresponding WeightedPodAffinityTerm). But say instead of using empty topologyKey, you create N separate soft anti-affinity rules, one for each of the keys that empty topologyKey maps to. Then that will count N times, i.e. one Weight for each of the N rules. Probably we want these to give the same overall weight.

So, I would suggest adding this TODO:

```
// TODO: One WeightedPodAfinityTerm with empty topologyKey adds Weight once to the priority,
// while N WeightedPodAffinityTerm, one for each of the N topologyKey that empty topologyKey
// maps to, adds N Weights to the priority. Ideally, these two scenarios should be weighted the same.
```